### ISSUE TITLE
The link “Back to previous page” leads back to a different page than the one I came from.

### ISSUE THREAD

**Issue by max951** [meeting-room-booking-system/mrbs-code#3662]
In the day view I select a date, let's say 28.04.2024 and click on an event to read the details.
If I then use the “Return to previous page” link, it takes me back to a different date. But that doesn't always happen.
The behavior occurs in Chrome and Firefox.

To illustrate this, I have [published a video](https://sirus.ziermaier.de/tmp/dgfc/Fehlfunktion-Booking.mp4) online in which the process can be seen.

I am using MRBS 1.11.4

**Comment by campbell-m** [comment]
I'm afraid I can't reproduce this.  Does it still happen on MRBS 1.11.5?  Also, what have you got `$auth['type']` and `$auth['session']` set to (the answers will be on the Help page if you log in to MRBS as an admin)?

**Comment by campbell-m** [comment]
OK.  What have you got `$auth['type']` and `$auth['session']` set to (the answers will be on the Help page if you log in to MRBS as an admin)?

**Comment by campbell-m** [comment]
> The following workaround seems to work at the moment:

Yes, but it could create other problems which `$referrer = session()->getReferrer();` was used to solve.

**Comment by max951** [comment]
Thank you very much for your time.

Oh, sorry, I forgot:

This is my config
```
MRBS version	MRBS 1.11.4
Databankschemaversion	82
Versions des lokalen Databankschemas	1
$auth['type']	dgfc
$auth['session']	php
```
This is my auth-file: (I have attached '.txt' to be able to upload the file)
[AuthDgfc.php.txt](https://github.com/meeting-room-booking-system/mrbs-code/files/15133067/AuthDgfc.php.txt)

**Comment by campbell-m** [comment]
Thanks.  Could you please

1. Check your PHP error log to see if you are getting any error messages when this problem happens.
2. Check your browser console for errors using your browser's developer tools.

**Comment by campbell-m** [comment]
By the way, you could simply your AuthDgfc.php code by extending AuthDbExt rather than AuthDb:

~~~php
namespace MRBS\Auth;


class AuthDgfc extends AuthDbExt
{
  private $db_moreconditions;
  
  public function __construct()
  {
    global $auth;
    
    parent::__construct();
    $this->db_moreconditions = $auth['db_ext']['db_moreconditions']; 
  }
  
  public function validateUser(
    #[\SensitiveParameter]
    ?string $user,
    #[\SensitiveParameter]
    ?string $pass)
  {
    // Your existing code here
  }
  
}
~~~

That way you can (a) cut down the amount of duplicated code and (b) benefit from enhancements and bug fixes to the duplicated code.

**Comment by campbell-m** [comment]
I've [now](https://github.com/meeting-room-booking-system/mrbs-code/commit/28daf2112e263e0aab69519d53287cd30fdf8889) separted some of the code in `validateUser()` so that in future releases of MRBS you can simplify your class still further.

**Comment by max951** [comment]
Thank you very much for your advice and your extensive work.

Unfortunately I can't find any hints about the behavior neither in the console nor in the PHP log. see screenshots

![hc_126](https://github.com/meeting-room-booking-system/mrbs-code/assets/12292845/ff2d7ed7-3e56-4a79-9364-3edd4304f9a6)
![hc_127](https://github.com/meeting-room-booking-system/mrbs-code/assets/12292845/61d37fa4-5324-4b98-ac6b-f69d19dc73ed)
